fearless/Common/Services/ExtrinsicService/ExtrinsicService.swift

```diff
@@ -4,17 +4,87 @@ import RobinHood
 import IrohaCrypto
 
 typealias ExtrinsicBuilderClosure = (ExtrinsicBuilderProtocol) throws -> (ExtrinsicBuilderProtocol)
-typealias EstimateFeeClosure = (Result<RuntimeDispatchInfo, Error>) -> Void
-typealias ExtrinsicSubmitClosure = (Result<String, Error>) -> Void
+typealias ExtrinsicBuilderIndexedClosure = (ExtrinsicBuilderProtocol, Int) throws -> (ExtrinsicBuilderProtocol)
+
+typealias FeeExtrinsicResult = Result<RuntimeDispatchInfo, Error>
+typealias EstimateFeeClosure = (FeeExtrinsicResult) -> Void
+typealias EstimateFeeIndexedClosure = (Result<[FeeExtrinsicResult], Error>) -> Void
```

Comment:
I think this look weird. I suggest to assume that if something failed before extrinsics are sent then assume that all extrinsics are failed and repeat error for all extrinsic results. Then the closure will be just `[Result<...>] -> Void`